### PR TITLE
add Axe (AXE)

### PR DESCRIPTION
**Pull request by charlesrocket**
Add AXE support

* https://axerunners.com

### CODE REVIEW

**Comment by charlesrocket** [comment]
`./scripts/build/docker/emulator/debug.sh` output:
```
[==========] Running 25 tests from 6 test cases.
[----------] Global test environment set-up.
[----------] 2 tests from Coins
[ RUN      ] Coins.Bip32PathToString
[       OK ] Coins.Bip32PathToString (3 ms)
[ RUN      ] Coins.TableSanity
[       OK ] Coins.TableSanity (0 ms)
[----------] 2 tests from Coins (3 ms total)

[----------] 1 test from Ethereum
[ RUN      ] Ethereum.AddressChecksum
[       OK ] Ethereum.AddressChecksum (0 ms)
[----------] 1 test from Ethereum (1 ms total)

[----------] 2 tests from Recovery
[ RUN      ] Recovery.ExactStrMatch
[       OK ] Recovery.ExactStrMatch (0 ms)
[ RUN      ] Recovery.AutoComplete
[       OK ] Recovery.AutoComplete (1 ms)
[----------] 2 tests from Recovery (1 ms total)

[----------] 17 tests from Storage
[ RUN      ] Storage.ReadMeta
[       OK ] Storage.ReadMeta (0 ms)
[ RUN      ] Storage.WriteMeta
[       OK ] Storage.WriteMeta (0 ms)
[ RUN      ] Storage.ReadPolicyV1
[       OK ] Storage.ReadPolicyV1 (0 ms)
[ RUN      ] Storage.WritePolicyV1
[       OK ] Storage.WritePolicyV1 (0 ms)
[ RUN      ] Storage.ReadStorageV1
[       OK ] Storage.ReadStorageV1 (6 ms)
[ RUN      ] Storage.WriteCacheV1
[       OK ] Storage.WriteCacheV1 (0 ms)
[ RUN      ] Storage.ReadCacheV1
[       OK ] Storage.ReadCacheV1 (0 ms)
[ RUN      ] Storage.DumpNode
[       OK ] Storage.DumpNode (0 ms)
[ RUN      ] Storage.ResetPolicies
[       OK ] Storage.ResetPolicies (0 ms)
[ RUN      ] Storage.SetPolicy
[       OK ] Storage.SetPolicy (0 ms)
[ RUN      ] Storage.ResetCache
[       OK ] Storage.ResetCache (0 ms)
[ RUN      ] Storage.StorageUpgrade_Normal
[       OK ] Storage.StorageUpgrade_Normal (0 ms)
[ RUN      ] Storage.StorageRoundTrip
[       OK ] Storage.StorageRoundTrip (5 ms)
[ RUN      ] Storage.UpgradePolicies
[       OK ] Storage.UpgradePolicies (0 ms)
[ RUN      ] Storage.IsPinCorrect
[       OK ] Storage.IsPinCorrect (0 ms)
[ RUN      ] Storage.Pin
[       OK ] Storage.Pin (0 ms)
[ RUN      ] Storage.Reset
[       OK ] Storage.Reset (0 ms)
[----------] 17 tests from Storage (12 ms total)

[----------] 1 test from USBRX
[ RUN      ] USBRX.Overflow
[       OK ] USBRX.Overflow (30 ms)
[----------] 1 test from USBRX (30 ms total)

[----------] 2 tests from U2F
[ RUN      ] U2F.WordsFromData
[       OK ] U2F.WordsFromData (0 ms)
[ RUN      ] U2F.ShapeShift
[       OK ] U2F.ShapeShift (0 ms)
[----------] 2 tests from U2F (0 ms total)

[----------] Global test environment tear-down
[==========] 25 tests from 6 test cases ran. (47 ms total)
[  PASSED  ] 25 tests.
Running main() from gtest_main.cc
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from Board
[ RUN      ] Board.Shutdown
[       OK ] Board.Shutdown (0 ms)
[ RUN      ] Board.U2FChannel
[       OK ] Board.U2FChannel (0 ms)
[----------] 2 tests from Board (1 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (2 ms total)
[  PASSED  ] 2 tests.
Running main() from gtest_main.cc
[==========] Running 4 tests from 2 test cases.
[----------] Global test environment set-up.
[----------] 2 tests from Crypto
[ RUN      ] Crypto.PermuteChar
[       OK ] Crypto.PermuteChar (0 ms)
[ RUN      ] Crypto.PermuteU16
[       OK ] Crypto.PermuteU16 (0 ms)
[----------] 2 tests from Crypto (1 ms total)

[----------] 2 tests from Vuln1845
[ RUN      ] Vuln1845.Bech32Decode
[       OK ] Vuln1845.Bech32Decode (0 ms)
[ RUN      ] Vuln1845.CashAddrDecode
[       OK ] Vuln1845.CashAddrDecode (0 ms)
[----------] 2 tests from Vuln1845 (0 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 2 test cases ran. (1 ms total)
[  PASSED  ] 4 tests.
Built target xunit
```

File: lib/firmware/coins.c [hunk 1]
```diff
@@ -68,6 +68,7 @@ const CoinType coins[COINS_COUNT] = {
...
     {true, "Zcash Testnet",true, "TAZ",   true, 7461,   true,   10000000,  true, 7354, false, 0,  false, 0,  true, "\x16" "Zcash Signed Message:\n",         true, 0x80000085,   false,   0,     true,   8,  false, {0, {0}},  false, {0, {0}},  true, 70617039,  true, 70615956,  false, false, false, false, true, SECP256K1_STRING,  false, "",            false, "",     false, false,  true, 63210096,   false,        0,        false,       0 },
+    {true, "Axe",          true, "AXE",   true,  55,    true,     100000,  true,  16,  false, 0,  false, 0,  true, "\x19" "DarkCoin Signed Message:\n",      true, 0x80001092,   false,   0,     true,   8,  false, {0, {0}},  false, {0, {0}},  true, 50221772,  true, 50221816,  true, false,  true, false,  true, SECP256K1_STRING,  false, "",            false, "",     false, false,  false, 0,         false,        0,        false,       0 },
```
Comment (keepkeyjon) [inline lib/firmware/coins.c]:
```suggestion
    {true, "Axe",          true, "AXE",   true,  55,    true,     100000,  true,  16,  false, 0,  false, 0,  true, "\x18" "AxeCoin Signed Message:\n",      true, 0x80001092,   false,   0,     true,   8,  false, {0, {0}},  false, {0, {0}},  true, 50221772,  true, 50221816,  true, false,  true, false,  true, SECP256K1_STRING,  false, "",            false, "",     false, false,  false, 0,         false,        0,        false,       0 },
```

File: include/keepkey/firmware/coins.h [hunk 2]
```diff
@@ -26,7 +26,7 @@
 #define ETHEREUM        "Ethereum"
 #define ETHEREUM_CLS    "Ethereum Classic"
 
-#define COINS_COUNT         60
+#define COINS_COUNT         61
```
Comment (keepkeyjon) [inline include/keepkey/firmware/coins.h]: Please squash this into the commit that adds the coin table row.

File: lib/firmware/coins.c [hunk 3]
```diff
@@ -257,11 +258,11 @@ const CoinType *coinBySlip44(uint32_t bip44_account_path)
 }
 
 /*
- * coin_amnt_to_str() - convert decimal coin amount to string for display 
+ * coin_amnt_to_str() - convert decimal coin amount to string for display
```
Comment (keepkeyjon) [inline lib/firmware/coins.c]: Please put these whitespace changes in their own separate commit.

**Comment by charlesrocket** [comment]
@keepkeyjon all done commits edited and squashed

how should I approach new conflicts?

**Comment by keepkeyjon** [comment]
> @keepkeyjon all done commits edited and squashed

Still need to fix the `signed_message_header` entry for Axe.

> how should I approach new conflicts?

rebase is fine.

**Comment by charlesrocket** [comment]
@keepkeyjon ow thanks! thats a big one 🤧

**Comment by keepkeyjon** [comment]
Rebased here: https://github.com/keepkey/keepkey-firmware/pull/59

**Pull request closed without merging**